easybuild/easyblocks/f/freefem.py

```diff
@@ -0,0 +1,165 @@
+            configopts.append('--with-hdf5=%s ' % os.path.join(hdf5_root, 'bin', 'h5pcc'))
+
+        gsl_root = get_software_root('GSL')
+        if gsl_root:
+            configopts.append('--with-gsl-prefix=%s' % gsl_root)
+
+        petsc_root = get_software_root('PETSc')
+        if petsc_root:
+            configopts.append('--with-petsc=%s' % os.path.join(petsc_root, 'lib', 'petsc', 'conf', 'petscvariables'))
+
+        bemtool_root = get_software_root('BemTool')
+        if bemtool_root:
+            configopts.append('--with-bem-include=%s' % bemtool_root)
+
+        for configopt in configopts:
+            self.cfg.update('configopts', configopt)
+
+        # initial configuration
+        out = super(EB_FreeFEM, self).configure_step()
+
+        regex = re.compile("WARNING: unrecognized options: (.*)", re.M)
+        res = regex.search(out)
+        if res:
+            raise EasyBuildError("One or more configure options not recognized: %s" % res.group(1))
+
+        if not petsc_root:
+            # re-create installation dir (deletes old installation),
+            # then set keeppreviousinstall to True (to avoid deleting PETSc installation)
+            self.make_installdir()
+            self.cfg['keeppreviousinstall'] = True
+
+            # configure and make petsc-slepc
+            # download & build PETSc as recommended by FreeFEM if no PETSc dependency was provided
+            cwd = change_dir(os.path.join('3rdparty', 'ff-petsc'))
+
+            cmd = ['make']
+            if self.cfg['parallel']:
+                cmd.append('-j %s' % self.cfg['parallel'])
+            cmd.append('petsc-slepc')
+
+            run_cmd(' '.join(cmd), log_all=True, simple=False)
+
+            change_dir(cwd)
+
+            # reconfigure for FreeFEM build
+            super(EB_FreeFEM, self).configure_step()
+
+    def test_step(self):
+        """Run tests."""
+
+        # run tests unless they're disabled explicitly
+        if self.cfg['runtest'] is None or self.cfg['runtest']:
+            # avoid oversubscribing, by using both OpenMP threads and having OpenBLAS use threads as well
+            env.setvar('OPENBLAS_NUM_THREAD', '1')
+
+            # avoid using too many OpenMP threads, which may slow tests down too much,
+            # since this could lead to failing tests (60 sec. time limit is imposed by the test run script)
+            parallel = self.cfg['parallel']
+            if parallel:
+                try:
+                    par = int(parallel)
+                except ValueError as err:
+                    raise EasyBuildError("Failed to parse 'parallel' value '%s' as integer: %s", parallel, err)
+
+                # use number of threads equal to 1/4th of available cores (and ensure at least one)
+                n_thr = max(par // 4, 1)
+            else:
+                n_thr = 1
+
+            env.setvar('OMP_NUM_THREADS', '%d' % n_thr)
+
+            run_cmd("make check", log_all=True, simple=False)
```

Comment:
I would make the installation fail if any of the tests fail or error out
```suggestion
            (out, _) = run_cmd("make check", log_all=True, simple=False, regexp=False)
                                                                                      
            # Raise an error if any test failed                                       
            check_log_for_errors(out, [('# (ERROR|FAIL): +[1-9]', ERROR)])            
```